test/auth.js

```diff
@@ -36,6 +72,52 @@ describe('auth interface', () => {
         .then((resp) => should(resp.uri).startWith('safe-auth:'));
   });
 
+  it('should throw error for non-existent permissions array for share MD request', () => {
+    const app = h.createTestApp();
+    const test = () => app.auth.genShareMDataUri();
+    should(test).throw(errConst.MISSING_PERMS_ARRAY.msg);
+  });
+
+  it('should throw error for misspelled or non-existent share MD request permission', async () => {
+    const app = h.createTestApp();
+    const sharedMdXorName = h.createRandomXorName();
+    const perms = [{ name: sharedMdXorName, perms: ['Insert'] }];
+    const test = () => app.auth.genShareMDataUri(perms);
+    should(test).throw(errConst.INVALID_SHARE_MD_PERMISSION.msg(JSON.stringify(perms[0])));
+  });
+
+  it('should throw error for malformed share MD request permission', async () => {
+    const app = h.createTestApp();
+    const sharedMdXorName = h.createRandomXorName();
+    const perms = { type_tag: 15001, name: sharedMdXorName, perms: ['Insert'] };
+    const test = () => app.auth.genShareMDataUri(perms);
+    should(test).throw(errConst.INVALID_PERMS_ARRAY.msg);
+  });
+
+  it('should throw error for invalid share MD request permission', async () => {
+    const app = h.createTestApp();
+    const sharedMdXorName = h.createRandomXorName();
+    const perms = [{ type_tag: 15001, name: sharedMdXorName, perms: ['Wrong'] }];
+    const test = () => app.auth.genShareMDataUri(perms);
+    should(test).throw(`${perms[0].perms[0]} is not a valid permission`);
+  });
+
+  it('should throw error for share MD request if type_tag is non-integer', async () => {
+    const app = h.createTestApp();
+    const sharedMdXorName = h.createRandomXorName();
+    const perms = [{ type_tag: 'non-integer', name: sharedMdXorName, perms: ['Insert'] }];
+    const test = () => app.auth.genShareMDataUri(perms);
+    should(test).throw('writeUInt64: no digits we found in input String');
+  });
+
+  it('should throw error for share MD request if name is not 32 byte buffer', async () => {
+    const app = h.createTestApp();
+    const mdName = 'not 32 byte buffer';
+    const perms = [{ type_tag: 15001, name: mdName, perms: ['Insert'] }];
+    const test = () => app.auth.genShareMDataUri(perms);
+    should(test).throw(`buffer length must be at least 32, got ${mdName.length}`);
```

Comment:
We had the code to return the proper error for this case: https://github.com/maidsafe/safe_app_nodejs/pull/208/files#diff-bdb8c45f3dc853fdcc4b43507d6ce996L146